### PR TITLE
Enhance the sturct of code

### PR DESCRIPTION
**Pull request by luckygalaxy666**
Adjust some structures and include of our codes to make calculation and  configuration methods in right files.

* Move all the functions like limit() into a new file, `mca_config.h`.
* Rename `mca.cpp` with `mca_config.cpp`
* Update the #include "mca.h" in `matrix.h` with #include "mca_config.h"
* Add #include "mca_config.h" in other files when the files use functions like limit()
* Move the functions, which is depended by the `matrix.h` in `single_thread_matrix_calculation.h`, into `matrix.h`

See #122

### CODE REVIEW

File: src/include/mca/__mca_internal/single_thread_matrix_calculation.h [hunk 1]
```diff
@@ -4,7 +4,6 @@
 #include <cassert>
 #include <cmath>
 #include <type_traits>
-
```
Comment (Kaiser-Yang) [inline src/include/mca/__mca_internal/single_thread_matrix_calculation.h:7]: Don't remove this line.

File: src/include/mca/matrix.h [hunk 2]
```diff
@@ -218,7 +286,37 @@ class Matrix {
      *                       [3, 4]] */
     inline Matrix transpose() const {
         Matrix<value_type> output(Shape{columns(), rows()});
-        mca::transpose(*this, output);
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:221]: Revert this change.

File: src/include/mca/matrix.h [hunk 3]
```diff
@@ -203,7 +261,17 @@ class Matrix {
     inline Matrix pow(const size_type &exponent) const {
         assert(isSquare());
         Matrix<value_type> output(shape());
-        mca::pow(*this, exponent, output);
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:206]: Revert this change.

File: src/include/mca/matrix.h [hunk 4]
```diff
@@ -185,7 +214,36 @@ class Matrix {
     template <class Number, class = std::enable_if_t<!is_matrix_v<Number>>>
     inline Matrix powNumber(const Number &number) {
         Matrix<value_type> output(shape());
-        mca::powNumber(*this, number, output);
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:188]: Revert this change.

File: src/include/mca/matrix.h [hunk 5]
```diff
@@ -166,10 +163,42 @@ class Matrix {
      *                  [2, 3, 4]]
      *              return: [[2^1, 2^2, 2^3],
      *                       [2^2, 2^3, 2^4]] */
+
+    
     template <class Number, class = std::enable_if_t<!is_matrix_v<Number>>>
     inline Matrix numberPow(const Number &number) {
         Matrix<value_type> output(shape());
-        mca::numberPow(number, *this, output);
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:172]: Revert this change.

File: src/include/mca/matrix.h [hunk 6]
```diff
@@ -166,10 +163,42 @@ class Matrix {
      *                  [2, 3, 4]]
      *              return: [[2^1, 2^2, 2^3],
      *                       [2^2, 2^3, 2^4]] */
+
+
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:167]: Revert this change.

File: src/include/mca/matrix.h [hunk 7]
```diff
@@ -11,13 +11,10 @@
 #include <utility>
 #include <vector>
 
-#include "__mca_internal/matrix_declaration.h"
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:14]: Revert this change.

File: src/include/mca/matrix.h [hunk 8]
```diff
@@ -11,13 +11,10 @@
...
-#include "__mca_internal/single_thread_matrix_calculation.h"
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:15]: Revert this change.

File: src/include/mca/matrix.h [hunk 9]
```diff
@@ -11,13 +11,10 @@
...
 #include "diag.h"
 #include "identity_matrix.h"
-#include "mca.h"
 #include "shape.h"
-
```
Comment (Kaiser-Yang) [inline src/include/mca/matrix.h:20]: Don't remove this line.

**Review by Kaiser-Yang — CHANGES_REQUESTED** [review]

**Review by Kaiser-Yang — APPROVED** [review]

**Review by Kaiser-Yang — APPROVED** [review]

**Review by Quentin9922 — APPROVED** [review]
